app/components/data-download.tsx

```diff
@@ -246,67 +237,91 @@ const DataDownloadInnerMenu = ({
   );
 };
 
+const DataDownloadMenuSection = ({
+  subheader,
+  dataToRender,
+  fileName,
+  onDownloaded,
+}: {
+  subheader: ReactNode;
+  dataToRender: PreparedData;
+  fileName: string;
+  onDownloaded: () => void;
+}) => {
+  return (
+    <>
+      <ListSubheader sx={{ mt: 3, lineHeight: 1.2, borderBottom: "none" }}>
+        {subheader}
+      </ListSubheader>
+      <MenuItem
+        sx={{
+          "&:hover": { backgroundColor: "transparent", cursor: "default" },
+        }}
+      >
+        <Flex sx={{ gap: 3 }}>
+          {FILE_FORMATS.map((fileFormat) => (
+            <DownloadMenuItem
+              key={fileFormat}
+              fileName={fileName}
+              fileFormat={fileFormat}
+              columnKeys={dataToRender.columnKeys}
+              data={dataToRender.data}
+              onDownloaded={onDownloaded}
+            />
+          ))}
+        </Flex>
+      </MenuItem>
+    </>
+  );
+};
+
 const DownloadMenuItem = ({
-  title,
+  fileName,
   data,
   columnKeys,
-  extent,
   fileFormat,
   onDownloaded,
 }: {
-  title: string;
+  fileName: string;
   data: Observation[];
   columnKeys: string[];
-  extent: Extent;
   fileFormat: FileFormat;
   onDownloaded: () => void;
 }) => {
   const [isDownloading, setIsDownloading] = useState(false);
   const download = useCallback(() => {
     return fetch("/api/download", {
       method: "POST",
-      headers: {
-        "Content-Type": "application/json",
-      },
-      body: JSON.stringify({
-        columnKeys,
-        data,
-        fileFormat,
-      }),
+      headers: { "Content-Type": "application/json" },
+      body: JSON.stringify({ columnKeys, data, fileFormat }),
     }).then((res) =>
-      res.blob().then((blob) => saveAs(blob, `${title}.${fileFormat}`))
+      res.blob().then((blob) => saveAs(blob, `${fileName}.${fileFormat}`))
     );
-  }, [columnKeys, data, fileFormat, title]);
+  }, [columnKeys, data, fileFormat, fileName]);
 
   return (
-    <MenuItem
+    <Button
+      variant="text"
+      size="small"
+      endIcon={isDownloading ? <CircularProgress /> : null}
       onClick={async () => {
         setIsDownloading(true);
         await download();
         onDownloaded();
         setIsDownloading(false);
       }}
-      sx={{ paddingY: 3 }}
+      sx={{
+        padding: 0,
+        minWidth: 0,
+        pointerEvents: isDownloading ? "none" : "auto",
```

Comment:
Instead of disabling pointerEvents, maybe the button itself should be disabled ?

Nit: For this usecase, I like to have a LoadingButton that receives a fetching prop (here it would receive "downloading" and adds the icon. I think since it is a generic usecase, it is good to have a generic component for this.

Nit: There is also the useDisclosure hook that gives you a way to easily use a boolean.

Reply:
I meant that this was a component that I like to have in my codebase, not that it was inside MUI. Although I see it in the Chinese version of the docs ? https://mui.com/zh/material-ui/api/loading-button/